### PR TITLE
Add privacy manifest

### PR DESCRIPTION
**Pull request by ffried**
### New Pull Request Checklist

- [x] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
- [x] I have read the [Documentation](http://cocoadocs.org/docsets/CocoaLumberjack/)
- [x] I have searched for a similar pull request in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/pulls) and found none

<br/>

- [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
- [x] I have added the required tests to prove the fix/feature I am adding
- [x] I have updated the documentation (if necessary)
- [x] I have run the tests and they pass
- [x] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / refers to the following issues: #1397 

### Pull Request Description

Adds a privacy manifest for the `CocoaLumberjack` target.

Open questions / points:

- [x] How will this integrate with CocoaPods?

### CODE REVIEW

**Comment by nabs-m** [comment]
> How will this integrate with CocoaPods?

@ffried The privacy manifest (along with its path) needs to be added to the `podspec`'s `resources` attribute. I don't see a `podspec` file in this repo, but I'm guessing it would be added to [this script](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/f44b9cb1e3f9d6d2108476ea28d336c196e943a6/Scripts/generate-podspec.sh)? Probably something like this under [both subspecs](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/f44b9cb1e3f9d6d2108476ea28d336c196e943a6/Scripts/generate-podspec.sh#L145-L153):
```
ss.resources = 'Sources/CocoaLumberjack/PrivacyInfo.xcprivacy'
```

[Here's an example](https://github.com/bugsnag/bugsnag-cocoa/pull/1616/files#diff-c4c40d3e49de76d8fac0fa28120520e20f1131fbca42210efb854a3dde7bb105R56-R58) from Bugsnag that I know worked well for consuming apps where the privacy manifest was properly imported.

**Comment by ffried** [comment]
@nabs-m Thanks a lot for this info! Saves me some research time!

Yea, we generate the pod spec on the fly to have one less file to manage.

I've added it to the script, though only to the Core subspec, since the Swift subspec has Core as its dependency. Thus I'd hope that this is picked up (works that way in SwiftPM). If not, I can always add it later on.

File: Scripts/generate-podspec.sh [hunk 1]
```diff
@@ -77,6 +77,16 @@ IOS_SDK="$(read_config_var "${IOS_SDK_CONFIG_VAR}" '[0-9]+\.[0-9]+' "${SDKS_XCCO
...
+SUPPORTED_SWIFT_VERSIONS=''
+for SPM_PKG_DEF in Package@swift-*.swift; do
+    SWIFT_VERSION="$(echo "${SPM_PKG_DEF}" | sed -E 's/^Package@swift-([0-9]+\.[0-9]+)\.swift$/\1/g')"
+    if [[ -n "${SWIFT_VERSION}" ]] && [[ "${SWIFT_VERSION}" != "${SPM_PKG_DEF}" ]]; then
+        # We add the comma to the end here, since we will add the last version at the end.
+        SUPPORTED_SWIFT_VERSIONS="${SUPPORTED_SWIFT_VERSIONS}'${SWIFT_VERSION}', "
+    fi
+done
+SUPPORTED_SWIFT_VERSIONS="${SUPPORTED_SWIFT_VERSIONS}'$(swift package tools-version | awk -F'.' '{print $1"."$2}')'"
+
```
Comment (nabs-m) [inline Scripts/generate-podspec.sh:89]: Nice 👌

**Review by nabs-m — APPROVED** [review]
Privacy manifest itself and the CocoaPods integration look good to me. I haven't tinkered with SPM integrations, but as long as it doesn't need to full path, it looks fine to me.

File: Scripts/generate-podspec.sh (hunk 1 above)
Comment (ffried) [reply Scripts/generate-podspec.sh:89]: Oh yeah. 🤠 
Don't know why I didn't to it like that in the first place...

**Comment by ffried** [comment]
> I haven't tinkered with SPM integrations, but as long as it doesn't need to full path, it looks fine to me.

That's the nice thing about SPM: It'll (usually) tell you if it doesn't find something. 🙃 
The reason it's not needed her is that it already knows the path to the Sources. And since the manifest is in the root of the sources directory for this target, no (sub-)path is needed.

**Comment by paulnicholson** [comment]
@ffried Thanks for your work on this! Is it ready to merge? How soon would it be released if it's merged? 
I don't think I can use a git reference for this in my podfile since the podspec is generated by a script.

**Comment by ffried** [comment]
@paulnicholson I've passed this along to my contact at Apple (that originally reached out to me - as mentioned in #1397) and am waiting for feedback.
I'd expect the enforcement to happen some time in April. We should be able to release a version containing these changes until then.

**Comment by layneios** [comment]
Hey team, sorry to tag here. Our app is using this CocoaLumberjack library, however, I got an error when generating Privacy Report. It shows me that missing an expected key: `NSPrivacyCollectedDataTypes` for this library. Could you make a fix if possible? 🙏

**Comment by ffried** [comment]
That's strange. According to the documentation, the API types is the only key necessary for third-party SDKs: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files

image

**Comment by ffried** [comment]
Interesting. I would have expected Xcode to verify the xcprivacy file of a target.
Either way, I added an empty list of collected data types, as well as `NSPrivacyTracking` with `NO`... That should do the trick.

As described in #1406, I'd release a new version towards the end of the week.

**Pull request merged**